### PR TITLE
feat: add trace nodes to SolveByElim

### PR DESCRIPTION
**Pull request by Vtec234**
Add trace nodes to `solve_by_elim` which make the search structure and local successes/failures a bit clearer. For example
screenshot

### CODE REVIEW

File: Mathlib/Tactic/SolveByElim.lean [hunk 1]
```diff
@@ -76,24 +76,32 @@ do
...
-    pure (lemmas' ++ ctx')
-
-  -- We attempt to find an expression which can be applied,
-  -- and for which all resulting sub-goals can be discharged using `solveByElim n`.
-  es.firstM fun e => do
-    trace[Meta.Tactic.solveByElim] "Trying to apply: {e}"
-    liftMetaTactic (fun mvarId => mvarId.apply e)
-    solveByElimAux lemmas ctx n
+  withTraceNode `Meta.Tactic.solveByElim
+    -- Note: the `addMessageContextFull` is so that the goal before any unification took place
```
Comment (gebner) [inline Mathlib/Tactic/SolveByElim.lean:90]: Yeah, that's a sore part of the API.  Sometimes you want to show the state afterwards, sometimes before.  If you have ideas on how to make this more ergonomic, let me know.

File: Mathlib/Tactic/SolveByElim.lean [hunk 2]
```diff
@@ -76,24 +76,32 @@ do
...
+    do
+      let es ← Elab.Term.TermElabM.run' do
+        let ctx' ← ctx
+        let lemmas' ← lemmas.mapM id
+        pure (lemmas' ++ ctx')
+
+      -- We attempt to find an expression which can be applied,
+      -- and for which all resulting sub-goals can be discharged using `solveByElim n`.
+      es.firstM fun e => withTraceNode `Meta.Tactic.solveByElim
+          (return m!"{exceptEmoji ·} trying to apply: {e}") do
```
Comment (dwrensha) [inline Mathlib/Tactic/SolveByElim.lean:102]: would "tried to apply" make more sense? It prints this *after* it knows whether it succeeded, right?

**Comment by dwrensha** [comment]
@Vtec234 usually mathlib4 PRs are from a branch on the main repo (due to some quirk in how the continuous integration works, I think). Do you have permission to add branches to the main mathlib4 repo?

File: Mathlib/Tactic/SolveByElim.lean (hunk 2 above)
Comment (Vtec234) [reply Mathlib/Tactic/SolveByElim.lean:102]: I'd say either one works and it's a matter of taste. It is true that the message is produced after running the tactic, but when reading the trace top-to-bottom you read the message *before* reading the nested ones corresponding to substeps. In any case, I changed it to `tried`.

**Pull request closed without merging**